VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitWebviewFragment.java

```diff
@@ -101,56 +112,82 @@ public void onActivityCreated(Bundle savedInstanceState) {
         webView.getSettings().setJavaScriptEnabled(true);
         URLInterceptorWebViewClient client =
                 new URLInterceptorWebViewClient(getActivity(), webView) {
-            @Override
-            public void onReceivedError(WebView view, int errorCode,
-                    String description, String failingUrl) {
-                hideLoadingProgress();
-                pageIsLoaded = false;
-                ViewPagerDownloadManager.instance.done(CourseUnitWebviewFragment.this, false);
-            }
-
-            // TODO: Restore these annotations when we upgrade our compile SDK version to Marshmallow
-            //@Override
-            //@TargetApi(Build.VERSION_CODES.M)
-            @TargetApi(23)
-            public void onReceivedHttpError(WebView view, WebResourceRequest request,
-                    WebResourceResponse errorResponse) {
-                switch (errorResponse.getStatusCode()) {
-                    case HttpURLConnection.HTTP_FORBIDDEN:
-                    case HttpURLConnection.HTTP_UNAUTHORIZED:
-                    case HttpURLConnection.HTTP_NOT_FOUND:
-                        EdxCookieManager.getSharedInstance().tryToRefreshSessionCookie();
-                        break;
-                }
-            }
-
-            public void onPageFinished(WebView view, String url) {
-                if (url != null && url.equals("data:text/html," + EMPTY_HTML)) {
-                    //we load a local empty html page to release the memory
-                } else {
-                    pageIsLoaded = true;
-                }
-
-                //TODO -disable it for now. as it causes some issues for assessment
-                //webview to fit in the screen. But we still need it to show additional
-                //compenent below the webview in the future?
-                // view.loadUrl("javascript:EdxAssessmentView.resize(document.body.getBoundingClientRect().height)");
-                ViewPagerDownloadManager.instance.done(CourseUnitWebviewFragment.this, true);
-                hideLoadingProgress();
-            }
-        };
+                    private boolean didReceiveError = false;
+
+                    @Override
+                    public void onReceivedError(WebView view, int errorCode,
+                                                String description, String failingUrl) {
+                        didReceiveError = true;
+                        hideLoadingProgress();
+                        pageIsLoaded = false;
+                        ViewPagerDownloadManager.instance.done(CourseUnitWebviewFragment.this, false);
+                        setFragmentState(true);
```

Comment:
> setFragmentState() method (which already is not best named) 

yeah, it definitely needs some improvements, but I deliberately put it up here to explore the options.

> The no_connectivity_layout layout can in that case be deleted, since the TextView can be defined in 2-3 lines.

Yeah, it seems like the better option.